### PR TITLE
Rasterized versions of ggbeeswarm

### PR DESCRIPTION
**Pull request by kpjonsson**
I thought this would be a neat implementation. Not sure it was the neatest way to do it though, I struggle with retaining the aspect ratios of rasterized points.

### CODE REVIEW

File: R/geom-beeswarm-rast.R [hunk 1]
```diff
@@ -0,0 +1,81 @@
+DrawGeomPointRast <- function(data, panel_params, coord, na.rm = FALSE, width=NULL, height=NULL, dpi=300) {
```
Comment (VPetukhov) [inline R/geom-beeswarm-rast.R]: Looks like this function is completely the same as in `geom_point_rast`. In this case, you don't need to redefine it.

File: R/geom-beeswarm-rast.R [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  cap <- grid::grid.cap()
+  dev.off(dev_id)
+  dev.set(prev_dev_id)
+
+  grid::rasterGrob(cap, x=0, y=0, width = 1,
+                   height = 1, default.units = "native",
+                   just = c("left","bottom"))
+}
+
+GeomPointRast <- ggplot2::ggproto(
```
Comment (VPetukhov) [inline R/geom-beeswarm-rast.R]: The same, you don't need to redefine this class.

File: R/geom-beeswarm-rast.R [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  groupOnX = NULL,
+  dodge.width = 0,
+  ...,
+  na.rm = FALSE,
+  show.legend = NA,
+  inherit.aes = TRUE,
+  width=NULL, height=NULL, dpi=300
+  ) {
+
+  position <- ggbeeswarm::position_beeswarm(priority = priority, cex = cex,
```
Comment (VPetukhov) [inline R/geom-beeswarm-rast.R]: Now your functions depend on `ggbeeswarm`. This means you need to add it to `Suggests` section of the `DESCRIPTION` file. Also, you need to check if this package is installed before using it (see [Hadley's guide](http://r-pkgs.had.co.nz/description.html#dependencies)).

File: R/geom-beeswarm-rast.R [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                   just = c("left","bottom"))
+}
+
+GeomPointRast <- ggplot2::ggproto(
+  "GeomPointRast",
+  ggplot2::GeomPoint,
+  draw_panel = DrawGeomPointRast
+)
+
+#' This geom is similar to \code{\link[ggplot2]{geom_point}}, but creates a raster layer
```
Comment (VPetukhov) [inline R/geom-beeswarm-rast.R]: Don't you mean `ggbeeswarm::position_beeswarm` here?

File: R/geom-beeswarm-rast.R [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+#' @inheritParams ggplot2::geom_point
+#' @inheritParams ggbeeswarm::position_beeswarm
+#' @inheritSection ggplot2::geom_point Aesthetics
+#'
+#' @param width Width of the result image (in inches). Default: deterined by the current device parameters.
+#' @param height Height of the result image (in inches). Default: deterined by the current device parameters.
+#' @param dpi Resolution of the result image.
+#'
+#' @examples
+#' ggplot() + geom_point_rast(aes(x=rnorm(1000), y=rnorm(1000)), dpi=600)
```
Comment (VPetukhov) [inline R/geom-beeswarm-rast.R]: Could you please add a valid example?

**Review by VPetukhov — CHANGES_REQUESTED** [review]
Hi, thank you very much for contribution!

Here are some comments on the code for `geom_beeswarm`. Could you please address them and the same issues for `geom_quasirandom`?

**Comment by kpjonsson** [comment]
@VPetukhov Thanks for the helpful comments. I think I've addressed this, can you my latest commit?

**Review by VPetukhov — APPROVED** [review]

**Comment by VPetukhov** [comment]
@kpjonsson Thank you for the quick fix! Merged your changes. There was a single issue, which you didn't fix: adding `ggbeeswarm` to Suggest section of the DESCRIPTION file, so I added it in f490890a6b7eac85bab6d45f5ff6bbea92781f88.

**Comment by kpjonsson** [comment]
Great, thanks!

**Pull request merged**